### PR TITLE
[Flaky Test][branch-1] TestFromClientSide#testCheckAndDeleteWithCompareOp

### PR DESCRIPTION
**Pull request by Reidddddd**
https://issues.apache.org/jira/browse/HBASE-25025

### CODE REVIEW

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 36s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 1 new or modified test files.  |
||| _ branch-1 Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   9m 53s |  branch-1 passed  |
| +1 :green_heart: |  compile  |   0m 42s |  branch-1 passed with JDK v1.8.0_262  |
| +1 :green_heart: |  compile  |   0m 45s |  branch-1 passed with JDK v1.7.0_272  |
| +1 :green_heart: |  checkstyle  |   1m 47s |  branch-1 passed  |
| +1 :green_heart: |  shadedjars  |   3m  3s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 48s |  branch-1 passed with JDK v1.8.0_262  |
| +1 :green_heart: |  javadoc  |   0m 40s |  branch-1 passed with JDK v1.7.0_272  |
| +0 :ok: |  spotbugs  |   3m  2s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   3m  0s |  branch-1 passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   1m 54s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 42s |  the patch passed with JDK v1.8.0_262  |
| +1 :green_heart: |  javac  |   0m 42s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 43s |  the patch passed with JDK v1.7.0_272  |
| +1 :green_heart: |  javac  |   0m 43s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   1m 36s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedjars  |   2m 49s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  hadoopcheck  |   4m 45s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2.  |
| +1 :green_heart: |  javadoc  |   0m 30s |  the patch passed with JDK v1.8.0_262  |
| +1 :green_heart: |  javadoc  |   0m 39s |  the patch passed with JDK v1.7.0_272  |
| +1 :green_heart: |  findbugs  |   2m 53s |  the patch passed  |
||| _ Other Tests _ |
| -1 :x: |  unit  | 134m 45s |  hbase-server in the patch failed.  |
| +1 :green_heart: |  asflicense  |   0m 37s |  The patch does not generate ASF License warnings.  |
|  |   | 176m 49s |   |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hbase.replication.TestReplicationSyncUpTool |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.12 Server=19.03.12 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2396/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/2396 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux 1df2c8c39b0d 4.15.0-60-generic #67-Ubuntu SMP Thu Aug 22 16:55:30 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-home/workspace/Base-PreCommit-GitHub-PR_PR-2396/out/precommit/personality/provided.sh |
| git revision | branch-1 / 7801b45 |
| Default Java | 1.7.0_272 |
| Multi-JDK versions | /usr/lib/jvm/zulu-8-amd64:1.8.0_262 /usr/lib/jvm/zulu-7-amd64:1.7.0_272 |
| unit | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2396/1/artifact/out/patch-unit-hbase-server.txt |
|  Test Results | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2396/1/testReport/ |
| Max. process+thread count | 4404 (vs. ulimit of 10000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2396/1/console |
| versions | git=1.9.1 maven=3.0.5 findbugs=3.0.1 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

**Comment by Reidddddd** [comment]
not related

File: hbase-server/src/test/java/org/apache/hadoop/hbase/client/TestFromClientSide.java [hunk 1]
```diff
@@ -5028,73 +5028,70 @@ public void testCheckAndDeleteWithCompareOp() throws IOException, InterruptedExc
...
+    assertFalse(ok);
+    // aaaa is less than bbbb, < || <= should be true
     ok = table.checkAndDelete(ROW, FAMILY, QUALIFIER, CompareOp.LESS, value1, delete);
-    assertEquals(ok, true);
-    table.put(put2);
+    assertTrue(ok);
+    table.put(bbbb);
     ok = table.checkAndDelete(ROW, FAMILY, QUALIFIER, CompareOp.LESS_OR_EQUAL, value1, delete);
-    assertEquals(ok, true);
-    table.put(put2);
```
Comment (virajjasani) [inline hbase-server/src/test/java/org/apache/hadoop/hbase/client/TestFromClientSide.java:5054]: so this was just a redundant put right?

File: hbase-server/src/test/java/org/apache/hadoop/hbase/client/TestFromClientSide.java [hunk 2]
```diff
@@ -5028,73 +5028,70 @@ public void testCheckAndDeleteWithCompareOp() throws IOException, InterruptedExc
...
-    put3.add(FAMILY, QUALIFIER, value3);
+    Put cccc = new Put(ROW);
+    cccc.add(FAMILY, QUALIFIER, value3);
 
     Delete delete = new Delete(ROW);
     delete.deleteColumns(FAMILY, QUALIFIER);
 
     // cell = "bbbb", using "aaaa" to compare only LESS/LESS_OR_EQUAL/NOT_EQUAL
     // turns out "match"
+    table.put(bbbb);
```
Comment (virajjasani) [inline hbase-server/src/test/java/org/apache/hadoop/hbase/client/TestFromClientSide.java:5042]: Curious to know if keeping this `put` operation on this line is a code improvement or has something to do with flaky behaviour?

Comment (Reidddddd) [reply hbase-server/src/test/java/org/apache/hadoop/hbase/client/TestFromClientSide.java:5042]: Just for readability only, this was originally in line 5033. I just separated it from initialization.

File: hbase-server/src/test/java/org/apache/hadoop/hbase/client/TestFromClientSide.java (hunk 1 above)
Comment (Reidddddd) [reply hbase-server/src/test/java/org/apache/hadoop/hbase/client/TestFromClientSide.java:5054]: Yes.

**Review by virajjasani — APPROVED** [review]
+1

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 47s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 1 new or modified test files.  |
||| _ branch-1 Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  10m 40s |  branch-1 passed  |
| +1 :green_heart: |  compile  |   0m 55s |  branch-1 passed with JDK v1.8.0_262  |
| +1 :green_heart: |  compile  |   1m  1s |  branch-1 passed with JDK v1.7.0_272  |
| +1 :green_heart: |  checkstyle  |   2m  0s |  branch-1 passed  |
| +1 :green_heart: |  shadedjars  |   3m  9s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 49s |  branch-1 passed with JDK v1.8.0_262  |
| +1 :green_heart: |  javadoc  |   0m 45s |  branch-1 passed with JDK v1.7.0_272  |
| +0 :ok: |  spotbugs  |   3m  5s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   3m  0s |  branch-1 passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   1m 55s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 39s |  the patch passed with JDK v1.8.0_262  |
| +1 :green_heart: |  javac  |   0m 39s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 44s |  the patch passed with JDK v1.7.0_272  |
| +1 :green_heart: |  javac  |   0m 44s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   1m 37s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedjars  |   2m 49s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  hadoopcheck  |   4m 39s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2.  |
| +1 :green_heart: |  javadoc  |   0m 30s |  the patch passed with JDK v1.8.0_262  |
| +1 :green_heart: |  javadoc  |   0m 39s |  the patch passed with JDK v1.7.0_272  |
| +1 :green_heart: |  findbugs  |   2m 52s |  the patch passed  |
||| _ Other Tests _ |
| -1 :x: |  unit  | 152m 14s |  hbase-server in the patch failed.  |
| +1 :green_heart: |  asflicense  |   0m 44s |  The patch does not generate ASF License warnings.  |
|  |   | 196m 22s |   |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hbase.mapreduce.TestImportExport |
|   | hadoop.hbase.replication.TestMasterReplication |
|   | hadoop.hbase.replication.TestReplicationStatus |
|   | hadoop.hbase.replication.regionserver.TestWALEntryStream |
|   | hadoop.hbase.util.TestHBaseFsck |
|   | hadoop.hbase.mapreduce.TestImportTsv |
|   | hadoop.hbase.replication.regionserver.TestRegionReplicaReplicationEndpointNoMaster |
|   | hadoop.hbase.mapreduce.TestMultiTableInputFormat |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.12 Server=19.03.12 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2396/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/2396 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux f17d911748d6 4.15.0-60-generic #67-Ubuntu SMP Thu Aug 22 16:55:30 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-home/workspace/Base-PreCommit-GitHub-PR_PR-2396/out/precommit/personality/provided.sh |
| git revision | branch-1 / 7801b45 |
| Default Java | 1.7.0_272 |
| Multi-JDK versions | /usr/lib/jvm/zulu-8-amd64:1.8.0_262 /usr/lib/jvm/zulu-7-amd64:1.7.0_272 |
| unit | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2396/2/artifact/out/patch-unit-hbase-server.txt |
|  Test Results | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2396/2/testReport/ |
| Max. process+thread count | 4370 (vs. ulimit of 10000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2396/2/console |
| versions | git=1.9.1 maven=3.0.5 findbugs=3.0.1 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

**Comment by Reidddddd** [comment]
This version should be much stable, let's see

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   6m 52s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 1 new or modified test files.  |
||| _ branch-1 Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  10m  0s |  branch-1 passed  |
| +1 :green_heart: |  compile  |   0m 44s |  branch-1 passed with JDK v1.8.0_262  |
| +1 :green_heart: |  compile  |   0m 48s |  branch-1 passed with JDK v1.7.0_272  |
| +1 :green_heart: |  checkstyle  |   1m 50s |  branch-1 passed  |
| +1 :green_heart: |  shadedjars  |   3m 13s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 51s |  branch-1 passed with JDK v1.8.0_262  |
| +1 :green_heart: |  javadoc  |   0m 45s |  branch-1 passed with JDK v1.7.0_272  |
| +0 :ok: |  spotbugs  |   3m 12s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   3m  9s |  branch-1 passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m  3s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 45s |  the patch passed with JDK v1.8.0_262  |
| +1 :green_heart: |  javac  |   0m 45s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 47s |  the patch passed with JDK v1.7.0_272  |
| +1 :green_heart: |  javac  |   0m 47s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   1m 42s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedjars  |   2m 59s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  hadoopcheck  |   4m 52s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2.  |
| +1 :green_heart: |  javadoc  |   0m 35s |  the patch passed with JDK v1.8.0_262  |
| +1 :green_heart: |  javadoc  |   0m 43s |  the patch passed with JDK v1.7.0_272  |
| +1 :green_heart: |  findbugs  |   3m  4s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  | 137m 22s |  hbase-server in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 39s |  The patch does not generate ASF License warnings.  |
|  |   | 187m 32s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.13 Server=19.03.13 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2396/3/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/2396 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux b4418eca196e 4.15.0-60-generic #67-Ubuntu SMP Thu Aug 22 16:55:30 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-home/workspace/Base-PreCommit-GitHub-PR_PR-2396/out/precommit/personality/provided.sh |
| git revision | branch-1 / 53d8af5 |
| Default Java | 1.7.0_272 |
| Multi-JDK versions | /usr/lib/jvm/zulu-8-amd64:1.8.0_262 /usr/lib/jvm/zulu-7-amd64:1.7.0_272 |
|  Test Results | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2396/3/testReport/ |
| Max. process+thread count | 4765 (vs. ulimit of 10000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2396/3/console |
| versions | git=1.9.1 maven=3.0.5 findbugs=3.0.1 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

**Pull request merged**